ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualSignedTransaction.kt

```diff
@@ -0,0 +1,79 @@
+ *  - It does not require a serializer.
+ *
+ * //TODO(below docs are from obsolete part, review them.)
+ * [ConsensualSignedTransaction] wraps the wire representation of the transaction. It contains one or more signatures,
+ * each one for a public key (including composite keys) that is mentioned inside a transaction state.
+ * [ConsensualSignedTransaction] is frequently passed around the network and stored. The identity of a transaction is
+ * the hash of Merkle root of the wrapped wire representation, therefore if you are storing data keyed by wire
+ * representations hash be aware that multiple different [ConsensualSignedTransaction]s may map to the same key (and
+ * they could be different in important ways, like validity!). The signatures on a ConsensualSignedTransaction
+ * might be invalid or missing: the type does not imply validity.
+ * A transaction ID should be the hash of the wrapped wire representation's Merkle tree root.
+ * Thus adding or removing a signature does not change it.
+ */
+@DoNotImplement
+@CordaSerializable
+interface ConsensualSignedTransaction {
+    /**
+     * @property id The ID of the transaction.
+     */
+    val id: SecureHash
+
+    /**
+     * @property signatures The signatures that have been applied to the transaction.
+     */
+    val signatures: List<DigitalSignatureAndMetadata>
+
+    /**
+     * Converts the current [ConsensualSignedTransaction] into a [ConsensualLedgerTransaction].
+     *
+     * @param serializer The [SerializationService] required to convert the current transaction.
+     * @return Returns a [ConsensualLedgerTransaction] from the current signed transaction.
+     */
+    fun toLedgerTransaction(serializer: SerializationService): ConsensualLedgerTransaction
+
+    /**
+     * Adds the specified signature to the current [ConsensualSignedTransaction].
+     *
+     * @param signature The signature and metadata to add to the current [ConsensualSignedTransaction].
+     * @return Returns a new [ConsensualSignedTransaction] containing the applied signature.
+     */
+    fun addSignature(signature: DigitalSignatureAndMetadata): ConsensualSignedTransaction
+
+    /**
+     * Adds the specified signatures to the current [ConsensualSignedTransaction].
+     *
+     * @param signatures The signatures and metadata to add to the current [ConsensualSignedTransaction].
+     * @return Returns a new [ConsensualSignedTransaction] containing the applied signatures.
+     */
+    fun addSignatures(signatures: Iterable<DigitalSignatureAndMetadata>): ConsensualSignedTransaction
+
+    /**
+     * Gets the signing keys for any missing transaction signatures.
+     *
+     * @param serializer The [SerializationService] required to obtain the missing keys.
+     * @return Returns a [Set] of [PublicKey] representing the signing keys for any missing transaction signatures.
+     */
+    fun getMissingSigningKeys(serializer: SerializationService): Set<PublicKey>
```

Comment:
The builder requires a list of consensual states to be provided before the transaction is generated, which are presumably not serialized until we generate the transaction itself. Can we extract a list of required signers at build time and prevent the need for a serialization round trip to extract them?

Reply:
Where the states are available the signers can be accessed too. (We planned to calculate them from the states's participants's owningKeys.)
I thought this function's purpose was more useful for verifying if there are any missing signatures.

Reply:
I have removed the serializer argument.